pkg/cli/config/config.go

```diff
@@ -96,6 +108,52 @@ func GetBackplaneConfiguration() (bpConfig BackplaneConfiguration, err error) {
 	return bpConfig, nil
 }
 
+func getConfigProxyList(proxyURLConfig any) ([]interface{}, error) {
+	// proxy-url in config file can be string or []string. Return the first working proxy.
+	var proxies []interface{}
+	var err error
+	switch t := proxyURLConfig.(type) {
+	case []interface{}:
+		logger.Infof("Multiple proxy URLs found in config file: %v ", t)
+		proxies = proxyURLConfig.([]interface{})
+	case interface{}:
+		proxies = []interface{}{proxyURLConfig}
+	}
+
+	return proxies, err
+}
+
+func (config *BackplaneConfiguration) getFirstWorkingProxyURL(s []interface{}) (string, error) {
```

Comment:
I believe this function has broken some other tests - in a good way. Now that we are verifying that proxy URLs are legitimate URLs, we need to update our tests to make sure they aren't using things like "example-proxy"

Reply:
I have refactored code to not throw error when no functioning proxy is found. This way existing functionality, of warning the user about proxy and moving on, is retained. All tests pass now. If we decide to throw error instead of warning, it breaks close to 20 tests.